Fast redirection after successfully saving an article. Don't re-enabl…

paulocheque opened the pull request:
…e the buttons in case of success. Also, renamed the Cancel button to Back.

paulocheque committed:
Fast redirection after successfully saving an article. Don't re-enable the buttons in case of success. Also, renamed the Cancel button to Back.

durden commented on pskb_website/static/js/editor_utils.js:
```diff
@@ -437,21 +437,25 @@ function save(sha, path, secondary_repo) {
         beforeSend: function(xhr) {
-            $('.btn-save').prop('disabled', true);
             $('html, body').css("cursor", "wait");
+            $('#editor-options').prop('disabled', true);
```
Good catch!  It's good to disable the options during send as well!

paulocheque committed:
Unselect all the lines after closing the live tutorial

paulocheque committed:
Added tooltip in the save button when the live tutorial is enabled

durden commented:
Any others ideas here or should I go ahead and merge?

paulocheque commented:
I was trying to think about disabling the button when the title was empty. But I did not find a elegant solution yet, because we need to consider other variables.

durden commented:
Ah ok, not a problem.  I just wanted to make sure it shouldn't be merged yet.

paulocheque committed:
Enable/Disable the save button if the user set the title and stack of the article

paulocheque commented:
Added a enable/disable function of the save button. It is fine to merge for me.

Test on `guides-dev`. Rebased with the `origin/master`. I can rebase with the `heroku/master` too.

durden commented:
I tested this a bit, then merged it.  Then, I found some issues so I created another PR to revert it.  So, the history, etc. is sort of messed up.  Guess there's no way to 're-open' a PR on github.  Either way, let me describe the issues and you can create another PR that fixes it.  Sorry for the confusion!

All the changes look good except the save button tooltip.  Maybe these issues have something to do with using `destroy` on the tooltip?

Here are the weird issues I've seen with the tooltip:
1. I cannot always get it to show up when I hover over the save button.
   - Try this to reproduce:
     - Immediately reload the page and hover over the save. This works.
     - Now reload the page and wait a few seconds **before** hovering over the save.  I cannot get the tooltip to show up until I hover once, hover away, then hover back.
2. Sometimes the tooltip on the save button gets stuck. It's hard to see in this screenshot, but my cursor is in the editor window but the save button tooltip is still visible.
   <img width="811" alt="screenshot 2016-05-05 10 04 37" src="https://cloud.githubusercontent.com/assets/58063/15039054/a5172732-12a9-11e6-9e80-756b3d3119f1.png">
3. Another example of the tooltip being stuck. I'm hovering over the options button but the save button tooltip is still visible:
   <img width="770" alt="screenshot 2016-05-05 10 11 38" src="https://cloud.githubusercontent.com/assets/58063/15039072/d6b485e6-12a9-11e6-9533-8364ffadfc29.png">
4. The code references a different tooltip on the save button when the live markdown preview is visible, but I cannot ever seem to see this tooltip at all when the markdown preview is displayed.

paulocheque commented:
Ok I will open a new PR for this.
I believe it is related to the disabled button. Tooltips does not work well with disabled buttons. But I will check the destroy part too.
